### ISSUE TITLE
`test_input_formats` failing on pypy3

### ISSUE THREAD

**Issue by atugushev** [jazzband/pip-tools#1375]
Introduced in #1356.

```console
2021-04-04T00:25:09.6159906Z         out = runner.invoke(cli, ["-n", "--find-links", dists_path, path])
2021-04-04T00:25:09.6160556Z >       assert out.exit_code == 0, out.stderr
2021-04-04T00:25:09.6161090Z E       AssertionError: 
2021-04-04T00:25:09.6161502Z E       assert 1 == 0
2021-04-04T00:25:09.6163174Z E        +  where 1 = <Result BackendUnavailable('Traceback (most recent call last):\n  File "/home/runner/work/pip-tools/pip-tools/.tox/pip... importlib._bootstrap>", line 962, in _find_and_load_unlocked\nModuleNotFoundError: No module named \'flit_core\'\n',)>.exit_code

...

2021-04-04T00:25:10.3688000Z >       assert out.exit_code == 0, out.stderr
2021-04-04T00:25:10.3688554Z E       AssertionError: 
2021-04-04T00:25:10.3689037Z E       assert 1 == 0
2021-04-04T00:25:10.3690850Z E        +  where 1 = <Result BackendUnavailable('Traceback (most recent call last):\n  File "/home/runner/work/pip-tools/pip-tools/.tox/pip...zen importlib._bootstrap>", line 962, in _find_and_load_unlocked\nModuleNotFoundError: No module named \'poetry\'\n',)>.exit_code
```

See [logs](https://github.com/jazzband/pip-tools/runs/2262017812?check_suite_focus=true) for details.

/cc: @orsinium

**Comment by webknjaz** [comment]
> No module named \'poetry\'\n',)

Looks like the trailing quote and line feed get into the name during parsing?

@atugushev why don't we run pypy jobs on PRs? They are not that heavy and being executed in parallel with the other jobs, the overall workflow completion would be barely noticeable.
Besides, I noticed that there's no caching set up in GHA, fixing this should make things even faster.

**Comment by webknjaz** [comment]
Oh, and if we ever get that "auto-merge" feature on, there will be even fewer reasons to care about the CI completion time. cc @jezdez

**Comment by atugushev** [comment]
@webknjaz 

We moved `pypy` to the cron workflow because they were too slow. However, there were `pypy2` jobs also at that time, but now there are only two jobs. It would be great to improve speed and move these jobs to PRs test workflow.

<img width="323" alt="image" src="https://user-images.githubusercontent.com/7377671/113543180-0a37bc00-9610-11eb-8752-429c0a7edb64.png">

**Comment by atugushev** [comment]
> Looks like the trailing quote and line feed get into the name during parsing?

It was just a bad repr (or a bug in `pytest`). Full traceback here:
```
Traceback (most recent call last):
  File "/Users/albert/Projects/pip-tools/tests/test_cli_compile.py", line 1773, in test_input_formats
    out = runner.invoke(cli, ["-n", "--find-links", dists_path, path], catch_exceptions=False)
  File "/Users/albert/Projects/pip-tools/.venv/site-packages/click/testing.py", line 329, in invoke
    cli.main(args=args or (), prog_name=prog_name, **extra)
  File "/Users/albert/Projects/pip-tools/.venv/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Users/albert/Projects/pip-tools/.venv/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/albert/Projects/pip-tools/.venv/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Users/albert/Projects/pip-tools/.venv/site-packages/click/decorators.py", line 21, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/Users/albert/Projects/pip-tools/piptools/scripts/compile.py", line 363, in cli
    dist = meta.load(os.path.dirname(os.path.abspath(src_file)))
  File "/Users/albert/Projects/pip-tools/.venv/site-packages/pep517/meta.py", line 71, in load
    path = Path(build_as_zip(builder))
  File "/Users/albert/Projects/pip-tools/.venv/site-packages/pep517/meta.py", line 58, in build_as_zip
    builder(dest=out_dir)
  File "/Users/albert/.pyenv/versions/pypy3.6-7.3.3/lib_pypy/_functools.py", line 80, in __call__
    return self._func(*(self._args + fargs), **fkeywords)
  File "/Users/albert/Projects/pip-tools/.venv/site-packages/pep517/meta.py", line 53, in build
    _prep_meta(hooks, env, dest)
  File "/Users/albert/Projects/pip-tools/.venv/site-packages/pep517/meta.py", line 28, in _prep_meta
    reqs = hooks.get_requires_for_build_wheel({})
  File "/Users/albert/Projects/pip-tools/.venv/site-packages/pep517/wrappers.py", line 169, in get_requires_for_build_wheel
    'config_settings': config_settings
  File "/Users/albert/Projects/pip-tools/.venv/site-packages/pep517/wrappers.py", line 275, in _call_hook
    raise BackendUnavailable(data.get('traceback', ''))
pep517.wrappers.BackendUnavailable: Traceback (most recent call last):
  File "/Users/albert/Projects/pip-tools/.venv/site-packages/pep517/in_process/_in_process.py", line 86, in _build_backend
    obj = import_module(mod_path)
  File "/Users/albert/.pyenv/versions/pypy3.6-7.3.3/lib-python/3/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1003, in _gcd_import
  File "<frozen importlib._bootstrap>", line 980, in _find_and_load
  File "<frozen importlib._bootstrap>", line 950, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1003, in _gcd_import
  File "<frozen importlib._bootstrap>", line 980, in _find_and_load
  File "<frozen importlib._bootstrap>", line 962, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'flit_core'
```

**Comment by orsinium** [comment]
Interesting. Looks like `build-system.requires` wasn't installed by pip.

**Comment by AndydeCleyre** [comment]
I was able to reproduce locally on Linux with (`pypy3.6-7.3.3`, `pypy3.7-7.3.2`, `pypy3.7-7.3.3`), though the problem did not occur when I additionally created a `requirements.in` file with the content: `.`, and `pip-compiled` against that.

So here's a pytest-less reproduction procedure:

```console
% mkdir -p workdir && cd workdir
% mkdir -p sample_lib && print -l "'example module'" "__version__ = '1.2.3'" >sample_lib/__init__.py
% eval "$(pyenv init - zsh)"
% pyenv install pypy3.6-7.3.3
% pyenv shell pypy3.6-7.3.3
% python -m venv venv
% . ./venv/bin/activate
% pip install -e PATH/TO/PIP-TOOLS
% $EDITOR pyproject.toml  # paste content from the flit test case
% pip-compile -f PATH/TO/PIP-TOOLS/tests/test_data/packages -f PATH/TO/PIP-TOOLS/tests/test_data/minimal_wheels pyproject.toml
```

It's not strictly necessary to specify the find-links or create the sample lib just to observe this, as the error is raised before any of those others would be anyway, I think.

It does look like `flit-core` gets installed, *somewhere*.

Notably, this succeeds:

```console
% pip install -f PATH/TO/PIP-TOOLS/tests/test_data/packages -f PATH/TO/PIP-TOOLS/tests/test_data/minimal_wheels .
```

Perhaps even more notably, the `pip-compile` command succeeds if I don't explicitly create a venv, but rather just use `pyenv`'s default activated venv for a pypy version. I also see though that `pip --version` reveals that the `pyenv` shim points to `/usr/lib/python3.9/site-packages/pip`. So I try:

```console
$ python -m pip install -U pip
```

Now the shim points to pypy's pip, but the command succeeds again, anyway.

**Comment by atugushev** [comment]
Cross reference https://github.com/pypa/pep517/issues/116

**Comment by atugushev** [comment]
Fixed by #1629
